### ISSUE TITLE
How to overwrite result of getSocketUrlParts?

### ISSUE THREAD

**Issue by Jack-Works** [pmmmwh/react-refresh-webpack-plugin#238]
https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/fd2568e2bc3eee1c8e4a9fa092a94e8321392a64/sockets/WDSSocket.js#L12-24

```js
function initWDSSocket(messageHandler, resourceQuery) {
  if (typeof __webpack_dev_server_client__ !== 'undefined') {
    const SocketClient = __webpack_dev_server_client__;

    const urlParts = getSocketUrlParts(resourceQuery);
    const connection = new SocketClient(url.format(urlParts));

    connection.onMessage(function onSocketMessage(data) {
      const message = JSON.parse(data);
      messageHandler(message);
    });
  }
}
```

Overlay now breaking in Chrome extension. On the extension page, it will calculate `urlParts` to `"chrome-extension://jkoeaghipilijlahjplgbfiocjhldnap/sockjs-node"`. In the next step `new SocketClient` it will throw.

```
Uncaught SyntaxError: The URL's scheme must be either 'http:' or 'https:'. 'chrome-extension:' is not allowed.
    at new SockJS (sockjs.js?9be2:683)
    at new SockJSClient (SockJSClient.js?0a33:43)
    at initWDSSocket (WDSSocket.js?b288:17)
    at Object.eval (ErrorOverlayEntry.js?e51d:76)
```

In the content script, it will throw this error because it is assuming webpack chunks are loaded by the \<script> tag (but actually not).

```
Uncaught TypeError: Cannot read property 'trim' of null
    at Module.m (webpack-internal:///./node_modules/native-url/dist/index.mjs:7)
    at getSocketUrlParts (webpack-internal:///./node_modules/@pmmmwh/react-refresh-webpack-plugin/sockets/utils/getSocketUrlParts.js:22)
    at initWDSSocket (webpack-internal:///./node_modules/@pmmmwh/react-refresh-webpack-plugin/sockets/WDSSocket.js:16)
```

If this line

https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/fd2568e2bc3eee1c8e4a9fa092a94e8321392a64/sockets/WDSSocket.js#L16

can be changed to

```js
    const urlParts = options.socketURL || getSocketUrlParts(resourceQuery);
```

Then I can fix the overlay in our environment.

**Comment by frehner** [comment]
#216  seems somewhat related, since I'm also trying to get that function to work for my use cases as well

**Comment by pmmmwh** [comment]
Fixed by #241
